app/api/getAbsences/route.ts

```diff
@@ -0,0 +1,83 @@
+import { prisma } from '../../../utils/prisma';
+import { NextResponse } from 'next/server';
+
+export async function GET() {
+  try {
+    const absences = await prisma.absence.findMany({
+      select: {
+        id: true,
+        lessonDate: true,
+        subject: true,
+        lessonPlan: true,
+        reasonOfAbsence: true,
+        absentTeacher: {
+          select: {
+            id: true,
+            firstName: true,
+            lastName: true,
+            email: true,
+          },
+        },
+        substituteTeacher: {
+          select: {
+            id: true,
+            firstName: true,
+            lastName: true,
+            email: true,
+          },
+        },
+        location: {
+          select: {
+            id: true,
+            name: true,
+            abbreviation: true,
+          },
+        },
+      },
+    });
+
+    if (!absences.length) {
+      return NextResponse.json({ events: [] }, { status: 200 });
+    }
+
+    const events = absences.map((absence) => {
+      function extractTimeArray(date: Date, offset = 0) {
+        const newDate = new Date(date);
+        newDate.setDate(date.getDate() + offset);
+        return [
+          newDate.getFullYear(),
+          newDate.getMonth() + 1,
+          newDate.getDate(),
+        ];
+      }
+      if (absence?.substituteTeacher) {
+        var substituteTeacherString = ` (${absence.substituteTeacher?.firstName} ${absence.substituteTeacher?.lastName[0]})`;
```

Comment:
Don't need the space
```suggestion
        var substituteTeacherString = `(${absence.substituteTeacher?.firstName} ${absence.substituteTeacher?.lastName[0]})`;
```

Reply:
@ludavidca